File: addressbook/java/Version_1.1/at.fh-joanneum.swd.pmt.addressbook.data/src/at/fh_joanneum/swd/pmt/addressbook/data/AddressDataStore.java
package at.fh_joanneum.swd.pmt.addressbook.data;

import java.util.Vector;

public class AddressDataStore implements IAddressDataStore {

	private Vector<Address> address;

	public AddressDataStore(){
		this.address = new Vector<Address>();		
	}
	
	@Override
	public boolean addAddress(Address newAddr) {
		if(findAddress(newAddr.getName()) == -1)
			return this.address.add(newAddr);
		else
			return false;
	}

	@Override
	public Address getAddressByName(String name) {		
		int idx = findAddress(name);
		if( idx>=0)
			return this.address.elementAt(idx);
		else
			return null;
	}

	@Override
	public Vector<Address> getAllAddresses() {
		return this.address;
	}

	@Override
	public boolean setAddress(String name, Address address) {
		int idx = findAddress(name);
		if(idx>=0){
			this.address.setElementAt(address, idx);
			return true;
		}else
			return false;
		
	}
	
	private int findAddress(String name){
		for(int i=0; i<this.address.size();i++){
			if(this.address.elementAt(i).getName().equalsIgnoreCase(name))
				return i;
		}
		return -1;
	}

	@Override
	public boolean removeAddress(String name) {
		int idx = findAddress(name);		
		if(idx>=0)
			return this.address.remove(this.address.get(idx));
		return false;
	}
}

File: addressbook/net/trunk/PMT.Addressbook.Data/AddressDataStore.cs
using System;
using System.Collections;
using System.Linq;
using System.Text;

namespace PMT.AddressBook.Data
{
    public class AddressDataStore
    {
        private static AddressDataStore instance;

        public AddressDataStore()
        {
            addresses = new ArrayList();
        }

        public static AddressDataStore Instance
        {
            get
            {
                if( instance == null )
                {
                    instance = new AddressDataStore();
                }
                return instance;
            }            
        }
        private ArrayList addresses;
        public ArrayList Addresses
        {
            get
            {
                return addresses;
            }            
        }

        public void addAddress(Address addr)
        {
            if(findAddress(addr.Name)== null)
                addresses.Add(addr);
        }

        public Address findAddress(String name)
        {
            foreach (Address i in addresses)
            {
                if (i.Name.CompareTo(name)==0)
                    return i;
            }
            return null;
        }

        public void removeAddress(String name)
        {
            Address tmp;
            tmp = findAddress(name);
            if(tmp != null)
                addresses.Remove(tmp);
        }
        public Address getElementAt(int idx)
        {
            if (addresses.Count < idx)
                return null;

            int cnt = 0;
            foreach (Address i in addresses)
            {
                if (cnt == idx)
                    return i;
            }
            return null;
        }
    }
}
